legal-api/src/legal_api/services/filings/validations/agm_extension.py

```diff
@@ -0,0 +1,109 @@
+# Copyright © 2023 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validation for the Agm Extension filing."""
+from http import HTTPStatus
+from typing import Dict, Optional
+
+from dateutil.relativedelta import relativedelta
+from flask_babel import _ as babel  # noqa: N813, I004, I001; importing camelcase '_' as a name
+# noqa: I003
+from legal_api.errors import Error
+from legal_api.models import Business
+from legal_api.utils.legislation_datetime import LegislationDatetime
+# noqa: I003
+
+
+def validate(business: Business, filing: Dict) -> Optional[Error]:
+    """Validate the AGM Extension filing."""
+    if not business or not filing:
+        return Error(HTTPStatus.BAD_REQUEST, [{'error': babel('A valid business and filing are required.')}])
+
+    msg = []
+
+    is_first_agm = filing['filing']['agmExtension']['isFirstAgm']
```

Comment:
since the schema is pretty loose and only really requires the year, I think we need to do some required validation checks for fields like `isFirstAgm`